include/tscore/PluginUserArgs.h

```diff
@@ -0,0 +1,70 @@
+/** @file
+
+  Base class and implementation details for the User Args features.
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#pragma once
+
+#include <array>
+#include "tscore/ink_assert.h"
+
+static constexpr int MAX_USER_ARGS_TXN   = 16;  /* max number of user arguments for Transactions */
```

Comment:
I wonder if this should be a `std::array` index by the enum values.

Reply:
Done.